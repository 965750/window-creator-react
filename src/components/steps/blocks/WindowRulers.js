import PropTypes from 'prop-types'
import React from 'react'
import styles from './WindowRulers.module.scss'

function WindowRulers({ window }) {
  let bottomRuler

  if (window.doorType === 1) {
    bottomRuler = (
      <div
        className={`${
          styles['ruler--horizontal']
        } absolute -bottom-25 md:-bottom-40 w-full border-l-2 border-scorpion border-r-2`}
      >
        <div
          className={`${styles.ruler__counter} absolute top-half left-half border bg-white text-center`}
        >
          <span>{window.width}</span>
        </div>
        <div className={`${styles['bar--horizontal']} w-full bg-scorpion`} />
      </div>
    )
  } else {
    bottomRuler = (
      <div className="flex">
        <div
          className={`${
            styles['ruler--horizontal']
          } absolute -bottom-25 md:-bottom-40 w-1/2 border-l-2 border-scorpion border-r-2`}
        >
          <div
            className={`${styles.ruler__counter} absolute top-half left-half border bg-white text-center`}
          >
            <span>{window.width}</span>
          </div>
          <div className={`${styles['bar--horizontal']} w-full bg-scorpion`} />
        </div>
        <div
          className={`${
            styles['ruler--horizontal']
          } absolute -bottom-25 md:-bottom-40 w-1/2 right-0 border-l-2 border-scorpion border-r-2`}
        >
          <div
            className={`${styles.ruler__counter} absolute top-half left-half border bg-white text-center`}
          >
            <span>{window.width}</span>
          </div>
          <div className={`${styles['bar--horizontal']} w-full bg-scorpion`} />
        </div>
      </div>
    )
  }

  return (
    <div>
      <div
        className={`${
          styles['ruler--vertical']
        } absolute -left-30 md:-left-40 h-full border-t-2 border-scorpion border-b-2`}
      >
        <div
          className={`${styles.ruler__counter} absolute top-half left-half border bg-white text-center`}
        >
          <span>{window.height}</span>
        </div>
        <div
          className={`${styles['bar--vertical']} h-full bg-scorpion mx-auto`}
        />
      </div>
      <div
        className={`${
          styles['ruler--horizontal']
        } absolute -top-25 md:-top-40 w-full border-l-2 border-scorpion border-r-2`}
      >
        <div
          className={`${styles.ruler__counter} absolute top-half left-half border bg-white text-center`}
        >
          <span>{window.doorType === 1 ? window.width : window.width * 2}</span>
        </div>
        <div className={`${styles['bar--horizontal']} w-full bg-scorpion`} />
      </div>
      {bottomRuler}
    </div>
  )
}

WindowRulers.propTypes = {
  window: PropTypes.shape({
    color: PropTypes.string,
    width: PropTypes.number,
    height: PropTypes.number,
    doorType: PropTypes.number,
    rows: PropTypes.array,
    columns: PropTypes.array,
  }),
}

export default WindowRulers
