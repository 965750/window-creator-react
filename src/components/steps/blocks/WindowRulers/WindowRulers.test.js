import '@testing-library/jest-dom/extend-expect'

import { applyMiddleware, compose, createStore } from 'redux';
import { IntlProvider } from 'react-intl'
import { Provider } from 'react-redux'
import { render, cleanup } from '@testing-library/react';

import WindowRulers from './index'
import messages from '../../../../messages-i18n'
import thunk from 'redux-thunk'
import React from 'react'

const initialStore = {
  theme: {
    lang: 'pl',
  },
}

const reducer = (state = initialStore, action) => {
  switch (action.type) {
    default:
      return state
  }
}

const store = createStore(
  reducer,
  compose(
    applyMiddleware(thunk),
  ),
)

const renderWithRedux = (component) => ({
  ...render(<Provider store={store}>{component}</Provider>),
})

describe('WindowRulers', () => {
  let wrapper
  const initialProps = {
    window: {
      color: '#000',
      width: 90,
      height: 180,
      doorType: 2,
      rows: [0, 1],
      columns: [0, 1],
    },
  }

  const setUp = (props = initialProps) => {
    wrapper = renderWithRedux(
      <IntlProvider
        locale="en"
        messages={messages[store.getState().theme.lang]}
      >
        <WindowRulers {...props} />
      </IntlProvider>,
    )
  }

  afterEach(cleanup)

  it('should match snapshot', () => {
    setUp()

    expect(wrapper.asFragment()).toMatchSnapshot()
  })

  it('should match when doorType is 1', () => {
    setUp({
      window: {
        ...initialProps.window,
        doorType: 1,
      }
    })

    expect(wrapper.asFragment()).toMatchSnapshot()
  })
})
