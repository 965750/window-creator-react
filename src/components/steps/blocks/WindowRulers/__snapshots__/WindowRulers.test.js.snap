// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WindowRulers should match snapshot 1`] = `
<DocumentFragment>
  <div>
    <div
      class="ruler--vertical absolute -left-30 md:-left-40 h-full border-t-2 border-scorpion border-b-2"
    >
      <div
        class="ruler__counter absolute top-half left-half border bg-white text-center"
      >
        <span>
          180
        </span>
      </div>
      <div
        class="bar--vertical h-full bg-scorpion mx-auto"
      />
    </div>
    <div
      class="ruler--horizontal absolute -top-25 md:-top-40 w-full border-l-2 border-scorpion border-r-2"
    >
      <div
        class="ruler__counter absolute top-half left-half border bg-white text-center"
      >
        <span>
          180
        </span>
      </div>
      <div
        class="bar--horizontal w-full bg-scorpion"
      />
    </div>
    <div
      class="flex"
    >
      <div
        class="ruler--horizontal absolute -bottom-25 md:-bottom-40 w-1/2 border-l-2 border-scorpion border-r-2"
      >
        <div
          class="ruler__counter absolute top-half left-half border bg-white text-center"
        >
          <span>
            90
          </span>
        </div>
        <div
          class="bar--horizontal w-full bg-scorpion"
        />
      </div>
      <div
        class="ruler--horizontal absolute -bottom-25 md:-bottom-40 w-1/2 right-0 border-l-2 border-scorpion border-r-2"
      >
        <div
          class="ruler__counter absolute top-half left-half border bg-white text-center"
        >
          <span>
            90
          </span>
        </div>
        <div
          class="bar--horizontal w-full bg-scorpion"
        />
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`WindowRulers should match when doorType is 1 1`] = `
<DocumentFragment>
  <div>
    <div
      class="ruler--vertical absolute -left-30 md:-left-40 h-full border-t-2 border-scorpion border-b-2"
    >
      <div
        class="ruler__counter absolute top-half left-half border bg-white text-center"
      >
        <span>
          180
        </span>
      </div>
      <div
        class="bar--vertical h-full bg-scorpion mx-auto"
      />
    </div>
    <div
      class="ruler--horizontal absolute -top-25 md:-top-40 w-full border-l-2 border-scorpion border-r-2"
    >
      <div
        class="ruler__counter absolute top-half left-half border bg-white text-center"
      >
        <span>
          90
        </span>
      </div>
      <div
        class="bar--horizontal w-full bg-scorpion"
      />
    </div>
    <div
      class="ruler--horizontal absolute -bottom-25 md:-bottom-40 w-full border-l-2 border-scorpion border-r-2"
    >
      <div
        class="ruler__counter absolute top-half left-half border bg-white text-center"
      >
        <span>
          90
        </span>
      </div>
      <div
        class="bar--horizontal w-full bg-scorpion"
      />
    </div>
  </div>
</DocumentFragment>
`;
